### PR TITLE
bugfixes to theory agnostic OOA

### PR DESCRIPTION
**Pull request by emanca**
1.  apply norm uncertainty to OOA splitting by charge
2. propagate helicity variations to OOA

### CODE REVIEW

File: wremnants/datasets/dataset_tools.py [hunk 1]
```diff
@@ -119,15 +119,16 @@ def makeFilelist(paths, maxFiles=-1, base_path=None, nano_prod_tags=None, is_dat
 
         filelist.extend(files)
 
+    toreturn = filelist if maxFiles < 0 or len(filelist) < maxFiles else random.Random(1).sample(filelist, maxFiles)
```
Comment (cippy) [inline wremnants/datasets/dataset_tools.py:122]: Why is this change needed ? I have the impression the result is the same

Comment (emanca) [reply wremnants/datasets/dataset_tools.py:122]: without this fix the option oneMCfileEveryN does nothing

**Comment by emanca** [comment]
It doesn't pass the CI because setupCombine is called with the ROOT interface instead of the hdf5 one that we use for the bias tests. In my opinion setupCombine in the CI should be run with the same setup as the bias tests. If @cippy knows why the ROOT interface fails and it's easily changeable I can deal with it in this PR otherwise it should be followed up separately

**Comment by cippy** [comment]
I mean, it says there is a list index out of range in a loop. It surely depends on how cardtool.addSystematics is configured. Since it was working before, it must be related to something you have modified recently. What I can see from the code is that either the argument formatWithValue is not set properly (it should be defined with as many entry as the number of axes for the systematics) or some other list appearing in that line doesn't have the proper number of entries. You can try running locally adding some printouts to understand what happens (can also use options -x ".*" -k ".*mass|.*yieldsTheory|.*norm" to only run on those systs you need for the debugging)

**Comment by cippy** [comment]
It might be that you define labelsByAxis=["PtVBin", "YVBin", "Helicity"], but left systAxesFlow=["ptVgenSig", "absYVgenSig"].
Their length must match

**Comment by bendavid** [comment]
My guess is that there is is some small difference in the way the loops are implemented such that in some cases if elements are missing from one list then the other list may be silent truncated.  (this is what happens when using zip over lists of different sizes for example)

In this case it would be good to understand this, since this condition may result in (silent) unintended behaviour when using the hdf5 output.

**Comment by cippy** [comment]
I think it is because you left mirror=True for the OOA, this will add a new axis to trigger a special behaviour when creating the mirrored images. I assume you don't want to mirror here (you are not mirroring for the in-acceptance bins I think)

**Comment by emanca** [comment]
> I think it is because you left mirror=True for the OOA, this will add a new axis to trigger a special behaviour when creating the mirrored images. I assume you don't want to mirror here (you are not mirroring for the in-acceptance bins I think). What I think happens is that this creates additional "entry" elements, but then the formatWithValue list might not be updated accordingly (I don't know exactly how that one works).

For the OOA we want mirror to be True because we don't pass the asymmetric bands explicitly

**Comment by emanca** [comment]
The fix I just pushed should solve the issue. The problem was that mirror=True was adding a new axis called "mirror" and it also needed to be appended to formatWithValue when used. This didn't happen in the hdf5 interface because that function isn't called and mirroring is done in a different way. Irrespective of that we should also migrate the CI to the hdf5 interface.

**Comment by bendavid** [comment]
Does this mean that different arguments would be needed to formatWithValue for root vs hdf5 output?

**Comment by cippy** [comment]
> Does this mean that different arguments would be needed to formatWithValue for root vs hdf5 output?

No, because the new element is added internally in CardTool, just like the "mirror" axis.

**Comment by cippy** [comment]
> The fix I just pushed should solve the issue. The problem was that mirror=True was adding a new axis called "mirror" and it also needed to be appended to formatWithValue when used. This didn't happen in the hdf5 interface because that function isn't called and mirroring is done in a different way. Irrespective of that we should also migrate the CI to the hdf5 interface.

As we discussed yesterday, until the hdf5 interface allows one to make debug checks on whatever output would be used in the fit, I think it is safer to keep using the current version. On the other hand CI could temporarily run both to ensure that they are in synch until the transition is complete (e.g. there can be a plot comparing impacts or even showing only the nuisances whose impacts have changed.
